src/data/error.rs

```diff
@@ -287,6 +324,45 @@ pub enum LexError {
     #[error("no newline at end of file")]
     NoNewlineAtEOF,
 
+    #[error("unknown token: {0:?}")]
+    UnknownToken(u8),
+
+    #[error("missing terminating \" character in string literal")]
+    StringMissingEndQuote,
```

Comment:
Do you want to combine this into `MissingEndQuote { string: bool }`? I think that would make it easier to create at the call site too.

Reply:
I chose to do separate errors because the error directive would have been large. I can still do it though.